pennylane/devices/task_qubit.py

```diff
@@ -0,0 +1,243 @@
+            TF and Torch datatypes.
+        future (None, bool): Indicates whether the internal circuit evaluation returns a future
+            to a result. This allows building of dependent workflows, but currently only works with
+            explicit calls to `device.batch_execute` with a PennyLane native device type such as
+            (`default.qubit`, `lightning.qubit`).
+        gen_report: Indicates whether the backend task-scheduler will generate a performance report based on the tasks that were run.
+    """
+
+    operations = DefaultQubit.operations
+    observables = DefaultQubit.observables
+
+    name = "Task-based proxy PennyLane plugin"
+    short_name = "task.qubit"
+    pennylane_requires = __version__
+    version = __version__
+    author = "Xanadu Inc."
+
+    supported_devices = {
+        "default.qubit",
+        "default.qubit.tf",
+        "default.qubit.jax",
+        "default.qubit.torch",
+        "lightning.qubit",
+    }
+
+    def __init__(
+        self,
+        wires,
+        *,
+        shots=None,
+        analytic=None,
+        backend="default.qubit",
+        gen_report: Union[bool, str] = False,
+        future=False,
+    ):
+        super().__init__(0, shots, cache=False)
+
+        self._backend = backend
+        self._backend_cls = qml.plugin_devices[backend].load()
+        self._backend_dev = self._backend_cls(wires=0)
+        self._wires = wires
+        self._gen_report = gen_report
+        self.num_wires = wires if isinstance(wires, int) else len(wires)
+        self._future = future
+        self.expand_fn = self.default_expand_fn
+
+        if backend not in TaskQubit.supported_devices:
+            raise Exception("Unsupported device backend.")
+
+    def __get__(self, obj, objtype=None):
+        if obj not in self.__dict__:
+            return self._backend_cls.obj
+        return self.__dict__[obj]
+
+    def batch_execute(self, circuits: List[qml.tape.QuantumTape]):
+        "This overloads the backt_execute functionality of QubitDevice to offload computations to a user-chosen backend device. This allows scaling of the available workers to instantiate a given number of backends for concurrent circuit evaluations."
+        if self._gen_report:
+            filename = self._gen_report if isinstance(self._gen_report, str) else "dask-report.html"
+            cm = performance_report(filename=filename)
+        else:
+            cm = nullcontext()
+
+        with cm:
+            results = []
+            if isinstance(circuits, dask.distributed.client.Future):
+                with worker_client() as client:
+                    results = client.submit(
+                        lambda backend, wires, tapes: [
+                            TaskQubit._execute_wrapper(backend, wires, i) for i in tapes
+                        ],
+                        self._backend,
+                        self._wires,
+                        circuits,
+                    )
```

Comment:
just so I'm following here; does Dask restrict you from calling instance methods on the client (e.g., so it's not as simply as doing `self.execute(tape)`, but instead you need to instantiate and execute a tape on the client manually?)